### PR TITLE
Add troubleshooting instructions for perpetual PIN entry case

### PR DESCRIPTION
**Pull request by xornivore**

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -84,3 +84,9 @@ env.sh: line 34: ${OS,,}: bad substitution
 ```
 
 Run `brew install bash`. The script is using a feature not that is not supported by the old macOS bash.
+
+## Key not recognized blocking code signing on PIN Entry
```
Comment (trishankatdatadog) [inline docs/troubleshooting.md:88]: Please link to this subsection from the ToC above.

File: gpg-gen-scdaemon-conf.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+# Stop on error.
+set -e
+
+# shellcheck disable=SC1091
+source env.sh
+
+echo "${GREEN}Generating scdaemon.conf."
```
Comment (trishankatdatadog) [inline gpg-gen-scdaemon-conf.sh:9]: I think we can add the `${RESET}` at the end of the line, no?

File: gpg-gen-scdaemon-conf.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
```
Comment (trishankatdatadog) [inline gpg-gen-scdaemon-conf.sh:1]: Could we:
1. rename this script to `scdaemon.sh`?
2. call this script from `gpg.sh`?

**Review by trishankatdatadog — CHANGES_REQUESTED** [review]

File: scdaemon.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+# Stop on error.
+set -e
+
+# shellcheck disable=SC1091
+source env.sh
```
Comment (daisukixci) [inline scdaemon.sh:7]:
```suggestion
[[ ${BASH_SOURCE[0]} != "$0" ]] || source env.sh
```

File: docs/troubleshooting.md [hunk 5]
```diff
@@ -5,6 +5,7 @@
 - [git rebase](#git-rebase)
 - [No PyUSB backend detected](#no-pyusb-backend-detected)
 - [Bad substitution](#bad-substitution)
+--[Card-not-recognized-with-perpetual-PIN-entry-prompt](#card-not-recognized-with-perpetual-pin-entry-prompt)
```
Comment (trishankatdatadog) [inline docs/troubleshooting.md:8]: Maybe call it `Unsupported device` or whatever the precise error message is, easier to find.

File: docs/troubleshooting.md [hunk 6]
```diff
@@ -84,3 +85,9 @@ env.sh: line 34: ${OS,,}: bad substitution
 ```
 
 Run `brew install bash`. The script is using a feature not that is not supported by the old macOS bash.
+
+## Card not recognized with perpetual PIN entry prompt
+
+This manifests as PIN Entry dialog prompting to insert the card in a perpetual loop.
+
+Run [gpg-gen-scdaemon-conf.sh](../scdaemon.sh), then `killall gpg-agent`.
```
Comment (trishankatdatadog) [inline docs/troubleshooting.md:93]:
```suggestion
Run [./scdaemon.sh](../scdaemon.sh).
```

Comment (trishankatdatadog) [reply docs/troubleshooting.md:93]: No `killall gpg-agent`. I will suggest something to add at the end of the script later.

File: scdaemon.sh [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+echo "${GREEN}Generating scdaemon.conf.${RESET}"
+
+# Sometimes on macOS, a gpg update make the yubikey detection flaky or completely impossible
+# So we enforce the scdaemon.conf configuration to detect the YubiKey as it is on macOS only
+# cf env.sh
+# https://gpgtools.tenderapp.com/discussions/problems/58454-after-updating-to-gpgtools-20171-yubikey-no-longer-functions-properly-both-in-mail-gpg2-card-edit/page/1
+if [[ -n "$SCDAEMON_CONF" ]]; then
+    backup_conf "$DEFAULT_GPG_SCDAEMON_CONF"
+    echo -e "$SCDAEMON_CONF" > "$DEFAULT_GPG_SCDAEMON_CONF"
+fi
```
Comment (trishankatdatadog) [inline scdaemon.sh:18]:
```suggestion
fi

$GPGCONF --kill all
```

**Review by trishankatdatadog — CHANGES_REQUESTED** [review]

File: docs/troubleshooting.md (hunk 5 above)
Comment (xornivore) [reply docs/troubleshooting.md:8]: There isn't an error message per se, it just doesn't see it.

**Review by trishankatdatadog — APPROVED** [review]
💯

**Pull request merged**